addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionManager.java

```diff
@@ -203,6 +203,18 @@ public void testManualPastMetadataBasedRule() throws InterruptedException {
         assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
     }
 
+    @Test
+    public void testManualNullMetadataBasedRule() throws InterruptedException {
+        RetentionRule testRule = createManualMetadataBasedRuleMillis("dc:expired", 500L);
+        Calendar haldSecond = Calendar.getInstance();
+        haldSecond.add(Calendar.MILLISECOND, -1000);
+        file = session.saveDocument(file);
+
+        file = service.attachRule(file, testRule, session);
+        assertTrue(file.isRecord());
+        assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
+    }
+
     @Test
     public void testRetainUntilDateSaved() throws InterruptedException {
         RetentionRule testRule = createManualImmediateRuleMillis(100L);
```

Comment:
`throw InterruptedException` is not needed

Reply:
It is actually, due to call to `awaitRetentionExpiration` that throws it.